internal/controller/controllers/installenv_controller.go

```diff
@@ -140,6 +144,40 @@ func (r *InstallEnvReconciler) updateClusterDiscoveryIgnitionIfNeeded(ctx contex
 	return nil
 }
 
+func (r *InstallEnvReconciler) processNMStateConfig(ctx context.Context, installEnv *adiiov1alpha1.InstallEnv) ([]*models.HostStaticNetworkConfig, error) {
+	var staticNetworkConfig []*models.HostStaticNetworkConfig
+
+	if installEnv.Spec.NMStateConfigLabelSelector.MatchLabels == nil {
+		return staticNetworkConfig, nil
+	}
+	NMStateLabelValue, found := installEnv.Spec.NMStateConfigLabelSelector.MatchLabels[SelectorNMStateConfigNameLabel]
```

Comment:
why is that a const? we can probably have multiple labels here no?

Reply:
Good question.
@mhrivnak, @avishayt,
Currently, this adds a label selector in InstallEnv spec and a label in NMStateConfig spec.
Do we expect a single InstallEnv to match with multiple NMstate labels?
alternately, do we expect a single NMStateConfig to match multiple InstallEnvs?

Reply:
Just at a high level, we still don't have consensus on the details of how to use nmstate in this API, so just be prepared for some flexibility.

> Do we expect a single InstallEnv to match with multiple NMstate labels?

Yes. As in static networking where IPs are assigned. Or different classes of hardware in the same environment require different configs.

> alternately, do we expect a single NMStateConfig to match multiple InstallEnvs?

No. Each environment defines its own configuration.

Reply:
Done.
both label name and value are now user-configurable.
CR examples updated to reflect that.